geps/gep-1897.md

```diff
@@ -125,7 +132,251 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+Version, e.g. ciphers mentioned for TLS version 1.1, 1.2, or 1.3 as described in 
+[RFC4346 TLS 1.1](https://datatracker.ietf.org/doc/html/rfc4346#appendix-A.5),
+[RFC 5246 TLS 1.2](https://datatracker.ietf.org/doc/html/rfc5246#appendix-A.5), or
+[RFC 8446 TLS 1.3](https://datatracker.ietf.org/doc/html/rfc8446#appendix-B.4).
+- Visibility: the visibility level could be `all` or `none`, and represents that if the connection failed, 
+the implmentation would drop the connection silently if the visibility level were `none`, or report an error if the
+visibility level were `all`.  (default to `all`)
+
+Thus, the following additions would be made to the Gateway API:
+
+```go
+// TLSConnectionPolicy provides a way to publish TLS configuration
+// that enables a gateway client to connect to a backend pod.
+type TLSConnectionPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of TLSConnectionPolicy.
+    Spec TLSConnectionPolicySpec `json:"spec"`
+
+    // Status defines the current state of TLSConnectionPolicy.
+    Status TLSConnectionPolicyStatus `json:"status,omitempty"`
+}
+
+// TLSConnectionPolicySpec defines the desired state of
+// TLSConnectionPolicy.
+// Note: there is no Override or Default policy configuration.
+//
+// Support: Core
+type TLSConnectionPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // TLS contains TLS connection policy configuration.
+    TLS *TLSConnectionPolicyConfig `json:”tls”`
+}
+
+// TLSConnectionPolicyConfig contains TLS connection policy configuration.
+type TLSConnectionPolicyConfig struct {
+    // TrustedCACertRefs contains one or more references to
+    // Kubernetes objects that contain PEM-encoded TLS certificates,
+    // which are used to establish a TLS handshake between the gateway
+    // and backend pod.
+    //
+    // If the TrustedCACertRefs is empty, then the system trusted
+    // certificates should be used. If there are none, or the
+    // implementation doesn't define system trusted certificates,
+    // then a TLS connection must fail.
+    //
+    // A single TrustedCACertRef to a Kubernetes ConfigMap has "Core"
+    // support.  Implementations MAY choose to support attaching
+    // multiple certificates to a backend, but this behavior is
+    // implementation-specific.
+    //
+    // References to a resource in a different namespace are 
+    // invalid.
+    // 
+    // Support: Core - An optional single reference to a Kubernetes
+    // ConfigMapName.
+    //
+    // Support: Implementation-specific (No reference, more than one
+    // reference, or other resource types.)
+    //
+    // +kubebuilder:validation:MaxItems=64
+    // +optional
+    TrustedCACertRefs []ConfigMapName `json:”trustedCACertRefs”`
```

Comment:
I think it's highly likely that in the future we'll support a different kind of resource to store certificates. To ensure we leave room for that, I'd rather use a type of reference that includes group and kind. We can/should still leave in the guidance that only ConfigMap has "Core" support here though.

Reply:
We may actually want to consider making [ClusterTrustBundle](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3257-trust-anchor-sets#clustertrustbundle-object) "Core" support in the near future whenever it graduates to beta or beyond. It would be good to reference that somewhere in this GEP as a future addition and maybe even state in this guidance that we anticipate transitioning to ClusterTrustBundle as it becomes more widely available.

Reply:
@robscott I need more context for the type of reference that includes group and kind, and still leave in support for ConfigMap.

Reply:
There's a a few ObjectReference types in `object_reference_types.go` in `v1beta1` - particularly LocalObjectReference may be useful there if we want to disallow cross-namespace CACertRefs. If we do want to allow cross-namespace references, probably the core k8s `ObjectReference` is the way to go (see https://pkg.go.dev/k8s.io/api/core/v1#ObjectReference)